pandas/tseries/base.py

```diff
@@ -69,6 +69,44 @@ def round(self, freq):
             values = self.tz_localize(None).asi8
         else:
             values = self.asi8
+        result = (unit*np.round(values/unit)).astype('i8')
+        attribs = self._get_attributes_dict()
+        if 'freq' in attribs:
+            attribs['freq'] = None
+        if 'tz' in attribs:
```

Comment:
pls reuse the existing code rather than repeating, e.g. something like:

```
def _round(self, freq, rounder):
        from pandas.tseries.frequencies import to_offset
         unit = to_offset(freq).nanos
 
         if getattr(self,'tz',None) is not None:
             values = self.tz_localize(None).asi8
         else:
             values = self.asi8
          result = (unit*rounder(values/unit)).astype('i8')
          attribs = self._get_attributes_dict()
          if 'freq' in attribs:
              result = result.tz_localize(self.tz)
          return result

def round(self, freq):
    return self._round(freq, np.round)

def floor(self, freq):
    return self._round(freq, np.floor)
```

furtther use an Appender on the doc-string to avoid repeating
```